flight/Modules/Battery/battery.c

```diff
@@ -128,18 +128,28 @@ static void batteryTask(void * parameters)
 				currentADCPin = -1;
 		}
 
+		bool adc_pin_invalid = false;
+		bool adc_offset_invalid = false;
+
 		// handle voltage
 		if (voltageADCPin >= 0) {
 			flightBatteryData.Voltage = (float)PIOS_ADC_GetChannelVolt(voltageADCPin);
 
 			// A negative result indicates an error (PIOS_ADC_GetChannelVolt returns -1 on error)
 			if(flightBatteryData.Voltage < 0.0f)
-				flightBatteryData.Voltage = 0.0f;
+				adc_pin_invalid = true;
```

Comment:
Hopefully the compiler will be smart enough to not use any more stack if I add a temporary variable, since the UAVO data struct is also in the local stack frame. I want to fit this module on the F1 targets very soon. ;-)